### ISSUE TITLE
Implement secret generation protocol

### ISSUE THREAD

**Issue by marsella** [boltlabs-inc/key-mgmt#132]
The cryptographic module requires several kinds of secrets to be [generated with a single protocol](https://github.com/boltlabs-inc/key-mgmt-spec/blob/main/cryptographic_flows.md#generate-a-secret).

The protocol itself is pretty straightforward. The most difficult part is to decide how to represent and store additional context when generating a secret. Different secrets will require different, overlapping contexts:
- An arbitrary `Secret` needs the user id and key id it's attached to
- A `StorageKey` might have other context, like a fixed string explaining what it is
- Both should specify the current version of key-mgmt

- [ ] Decide (and log) any additional context that isn't noted here
- [ ] Generate an appropriate abstraction to define different contexts
- [ ] Implement protocol

**Comment by indomitableSwan** [comment]
Related to https://github.com/boltlabs-inc/key-mgmt-spec/issues/56.

**Comment by indomitableSwan** [comment]
Forward-looking thoughts: 
- For user-facing arbitrary secret types: 
    - We may want to be able to handle a user-set fixed string as context, in addition to a default context string (e.g., key id, user id), particularly for an arbitrary secret that _needs to be stored_ at the database. As mentioned in https://github.com/boltlabs-inc/key-mgmt-spec/issues/56. 
    - In the above, I'm hinting that we may have other types of user-facing arbitrary secrets that _do not_ get stored in a server database, and are instead re-derived deterministically at time of use. 
- I have probably commented on this before, but I think we want a low-level abstraction for an arbitrary secret (a type that is _not_ public) that we can use under the hood to build up user-facing key types, where this type includes an arbitrary string (set at generation, provided as an input) as context.
- TLDR: I think it's ok to do something simple right now and refactor with more appropriate abstractions later. Add contexts back into the `Secret` struct (or I guess that looks like it may have already happened [here](https://github.com/boltlabs-inc/key-mgmt/pull/151/commits/8e8c0c993d227b165d7f348b51c27cb6f7800af3)) and take a context string as input to generate.
